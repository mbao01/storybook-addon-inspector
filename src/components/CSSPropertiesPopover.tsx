"use client";

import React from "react";

import { useState } from "react";
import {
  ChevronDown,
  ChevronRight,
  Code,
  Paintbrush,
  Variable,
} from "lucide-react";

import { Popover, PopoverContent } from "../ui/popover";
import { ScrollArea } from "../ui/scroll-area";
import { Separator } from "../ui/separator";
import { Badge } from "../ui/badge";
import { cn } from "../lib/utils";

type CSSPropertyValue = Record<
  "value" | "token" | "computed" | "variable" | "variableValue",
  string | undefined
> & {
  property: string;
};

type Data = {
  tokens: CSSPropertyValue[];
  computed: CSSPropertyValue[];
  variables: CSSPropertyValue[];
};

type CSSPropertyPopoverProps = {
  id: string;
  data: Data;
  open: boolean;
};

export const CSSPropertiesPopover = ({
  id,
  data,
  open,
}: CSSPropertyPopoverProps) => {
  const [isExpanded, setExpanded] = useState(true);

  const tokens = data.tokens ?? [];
  const computed = data.computed ?? [];
  const variables = data.variables ?? [];

  const hasNoProperties =
    tokens.length === 0 && variables.length === 0 && computed.length === 0;

  return (
    <Popover open={open}>
      <PopoverContent
        id={id}
        className={cn(
          "fixed top-9 right-1 z-50 w-[350px] p-0 bg-white dark:bg-gray-900 rounded-tl-none!",
          {
            "border-gray-200 dark:border-gray-800": isExpanded,
            "w-[172px]": !isExpanded,
          },
        )}
      >
        <div className="relative">
          <div className="absolute -top-7.5 left-0 right-4 flex items-center w-[172px]">
            <button
              type="button"
              onClick={() => setExpanded((e) => !e)}
              className="bg-white dark:bg-gray-900 text-gray-900/60 dark:text-white/60 px-3 py-1.5 -ml-px text-xs font-medium rounded-t-lg border border-gray-200 dark:border-gray-800 border-b-0 cursor-pointer"
            >
              CSS Property Inspector{" "}
              {isExpanded ? (
                <ChevronDown className="size-4 text-gray-500 dark:text-gray-400 flex-shrink-0" />
              ) : (
                <ChevronRight className="size-4 text-gray-500 dark:text-gray-400 flex-shrink-0" />
              )}
            </button>
          </div>

          {isExpanded && (
            <div className="relative h-[400px]">
              <ScrollArea className="h-full pr-3">
                <div className="space-y-1">
                  {tokens.length > 0 && (
                    <PropertySection
                      title="Tokens"
                      icon={<Paintbrush className="size-4 text-emerald-500" />}
                      properties={tokens}
                    />
                  )}

                  {variables.length > 0 && (
                    <>
                      {tokens.length > 0 && (
                        <Separator className="my-1 bg-gray-200 dark:bg-gray-800" />
                      )}
                      <PropertySection
                        title="Variables"
                        icon={<Variable className="size-4 text-violet-500" />}
                        properties={variables}
                      />
                    </>
                  )}

                  {computed.length > 0 && (
                    <>
                      {(tokens.length > 0 || variables.length > 0) && (
                        <Separator className="my-1 bg-gray-200 dark:bg-gray-800" />
                      )}
                      <PropertySection
                        title="Computed"
                        icon={<Code className="size-4 text-sky-500" />}
                        properties={computed}
                      />
                    </>
                  )}

                  {hasNoProperties && (
                    <div className="py-6 text-sm text-center text-gray-500 dark:text-gray-400">
                      No CSS properties found
                    </div>
                  )}
                </div>
              </ScrollArea>
            </div>
          )}
        </div>
      </PopoverContent>
    </Popover>
  );
};

type PropertySectionProps = {
  title: string;
  icon: React.ReactNode;
  properties: CSSPropertyValue[];
};

const PropertySection = ({ title, icon, properties }: PropertySectionProps) => {
  const [isExpanded, setExpanded] = useState(true);
  return (
    <div className="w-full">
      <button
        onClick={() => setExpanded((e) => !e)}
        className="sticky top-0 z-10 bg-white dark:bg-gray-900 w-full px-4 py-2 flex items-center gap-2 cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-800 text-left"
        type="button"
      >
        {isExpanded ? (
          <ChevronDown className="size-4 text-gray-500 dark:text-gray-400 flex-shrink-0" />
        ) : (
          <ChevronRight className="size-4 text-gray-500 dark:text-gray-400 flex-shrink-0" />
        )}
        {icon}
        <h3 className="font-medium text-gray-900 dark:text-white">{title}</h3>
        <Badge
          variant="outline"
          className="ml-auto border-gray-200 dark:border-gray-700 text-gray-700 dark:text-gray-300"
        >
          {properties.length}
        </Badge>
      </button>

      {isExpanded && (
        <div className="space-y-2 px-4 py-2">
          {properties.map((prop, index) => (
            <PropertyItem key={`${prop.property}-${index}`} property={prop} />
          ))}
        </div>
      )}
    </div>
  );
};

const PropertyItem = ({ property }: { property: CSSPropertyValue }) => {
  return (
    <div className="rounded-md border border-gray-200 dark:border-gray-800 p-2 text-sm bg-white dark:bg-gray-900">
      <div className="font-medium text-gray-900 dark:text-white">
        {property.property}
      </div>
      <div className="grid grid-cols-2 gap-x-2 gap-y-1 mt-1 text-xs">
        {property.value && (
          <>
            <span className="text-gray-500 dark:text-gray-400">Value:</span>
            <ColorValue value={property.value} />
          </>
        )}
        {property.token && (
          <>
            <span className="text-gray-500 dark:text-gray-400">Token:</span>
            <span className="font-mono text-emerald-600 dark:text-emerald-500">
              {property.token}
            </span>
          </>
        )}
        {property.variable && (
          <>
            <span className="text-gray-500 dark:text-gray-400">Variable:</span>
            <span className="font-mono text-violet-600 dark:text-violet-500">
              {property.variable}
            </span>
          </>
        )}
        {property.variableValue && (
          <>
            <span className="text-gray-500 dark:text-gray-400">
              Variable Value:
            </span>
            <ColorValue value={property.variableValue} />
          </>
        )}
        {property.computed && (
          <>
            <span className="text-gray-500 dark:text-gray-400">Computed:</span>
            <ColorValue value={property.computed} />
          </>
        )}
      </div>
    </div>
  );
};

const ColorValue = ({ value }: { value: string }) => {
  const isColor = isValidColor(value);

  return (
    <div className="flex items-center gap-1.5">
      {isColor && (
        <div
          className="size-3.5 rounded-sm border border-gray-300 dark:border-gray-700 flex-shrink-0"
          style={{
            backgroundColor: value,
            boxShadow: "inset 0 0 0 1px rgba(0,0,0,0.05)",
          }}
        />
      )}
      <span
        className={`font-mono ${isColor ? "flex-1" : ""} text-gray-900 dark:text-gray-100`}
      >
        {value}
      </span>
    </div>
  );
};

// Helper function to check if a string is a valid CSS color
const isValidColor = (value: string): boolean => {
  // Check for hex colors (3, 4, 6, or 8 digits)
  if (/^#([0-9A-F]{3,4}|[0-9A-F]{6}|[0-9A-F]{8})$/i.test(value)) {
    return true;
  }

  // Check for rgb/rgba colors
  if (
    /^rgba?$$\s*\d+\s*,\s*\d+\s*,\s*\d+\s*(?:,\s*(?:0?\.\d+|[01])\s*)?$$$/i.test(
      value,
    )
  ) {
    return true;
  }

  // Check for hsl/hsla colors
  if (
    /^hsla?$$\s*\d+\s*,\s*\d+%\s*,\s*\d+%\s*(?:,\s*(?:0?\.\d+|[01])\s*)?$$$/i.test(
      value,
    )
  ) {
    return true;
  }

  // Check for named colors (simplified check - we're just checking for common color names)
  const namedColors = [
    "black",
    "white",
    "red",
    "green",
    "blue",
    "yellow",
    "purple",
    "orange",
    "pink",
    "brown",
    "gray",
    "grey",
    "cyan",
    "magenta",
    "lime",
    "olive",
    "navy",
    "teal",
    "aqua",
    "silver",
    "gold",
    "indigo",
    "violet",
  ];

  return namedColors.some((color) => value.toLowerCase() === color);
};
