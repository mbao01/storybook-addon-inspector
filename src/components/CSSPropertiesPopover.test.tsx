import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { CSSPropertiesPopover } from "./CSSPropertiesPopover";

// Mock the Popover component to avoid portal issues
vi.mock("../ui/popover", () => ({
  Popover: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  PopoverContent: ({ children, className, id }: { children: React.ReactNode, className?: string, id?: string }) => (
    <div id={id} className={className} data-testid="popover-content">
      {children}
    </div>
  ),
}));

describe("CSSPropertiesPopover", () => {
  const mockProps = {
    id: "test-popover",
    tokens: [
      {
        property: "color",
        value: "red",
        token: "COLOR_PRIMARY",
        computed: "rgb(255, 0, 0)",
        variable: undefined,
        variableValue: undefined,
      },
    ],
    computed: [
      {
        property: "font-size",
        value: "16px",
        token: undefined,
        computed: "16px",
        variable: undefined,
        variableValue: undefined,
      },
    ],
    variables: [
      {
        property: "background-color",
        value: "var(--bg-color)",
        token: undefined,
        computed: "rgb(0, 0, 0)",
        variable: "--bg-color",
        variableValue: "rgb(0, 0, 0)",
      },
    ],
    open: true,
  };

  it("renders with all property groups", () => {
    const { asFragment } = render(<CSSPropertiesPopover {...mockProps} />);

    // Check section titles
    expect(screen.getByText("Tokens")).toBeInTheDocument();
    expect(screen.getByText("Variables")).toBeInTheDocument();
    expect(screen.getByText("Computed")).toBeInTheDocument();

    // Check property values
    expect(screen.getByText("color")).toBeInTheDocument();
    expect(screen.getByText("font-size")).toBeInTheDocument();
    expect(screen.getByText("background-color")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders empty state when no properties are provided", () => {
    render(
      <CSSPropertiesPopover
        id="test-popover"
        tokens={[]}
        computed={[]}
        variables={[]}
        open={true}
      />,
    );

    expect(screen.getByText("No CSS properties found")).toBeInTheDocument();
  });

  it("toggles expansion state when clicking the header button", () => {
    render(<CSSPropertiesPopover {...mockProps} />);

    const headerButton = screen.getByText("CSS Property Inspector");
    fireEvent.click(headerButton);

    // After clicking, the content should be hidden
    expect(screen.queryByText("Tokens")).not.toBeInTheDocument();

    // Click again to expand
    fireEvent.click(headerButton);
    expect(screen.getByText("Tokens")).toBeInTheDocument();
  });

  it("renders color values with color swatches", () => {
    render(<CSSPropertiesPopover {...mockProps} />);

    // Check for color swatch
    const colorSwatch = screen.getByTestId("color-swatch");
    expect(colorSwatch).toBeInTheDocument();
  });

  it("renders property sections with correct badges", () => {
    render(<CSSPropertiesPopover {...mockProps} />);

    // Check badge counts - use getAllByText and check length
    const badges = screen.getAllByText("1");
    expect(badges.length).toBeGreaterThanOrEqual(3); // At least 3 badges (one for each section)
  });

  it("toggles section expansion when clicking section headers", () => {
    render(<CSSPropertiesPopover {...mockProps} />);

    // Click on the Tokens section header
    const tokensHeader = screen.getByText("Tokens").closest("button");
    fireEvent.click(tokensHeader!);

    // The color property should be hidden
    expect(screen.queryByText("color")).not.toBeInTheDocument();

    // Click again to expand
    fireEvent.click(tokensHeader!);
    expect(screen.getByText("color")).toBeInTheDocument();
  });

  it("renders with correct CSS classes based on expansion state", () => {
    render(<CSSPropertiesPopover {...mockProps} />);

    // Initially expanded
    const popoverContent = screen.getByTestId("popover-content");
    expect(popoverContent).toHaveClass("ia:w-[350px]");

    // Click to collapse
    const headerButton = screen.getByText("CSS Property Inspector");
    fireEvent.click(headerButton);

    // Should have collapsed width
    expect(popoverContent).toHaveClass("ia:w-[172px]");
  });
});