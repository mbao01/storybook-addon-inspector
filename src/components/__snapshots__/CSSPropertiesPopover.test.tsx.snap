// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSSPropertiesPopover > renders with all property groups 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ia:fixed ia:top-9 ia:right-1 ia:z-50 ia:w-[350px] ia:p-0 ia:bg-white ia:dark:bg-gray-900 ia:rounded-tl-none! ia:border-gray-200 ia:dark:border-gray-800"
      data-testid="popover-content"
      id="test-popover"
    >
      <div
        class="ia:relative"
      >
        <div
          class="ia:absolute ia:-top-7.5 ia:left-0 ia:right-4 ia:flex ia:items-center ia:w-[172px]"
        >
          <button
            class="ia:bg-white ia:dark:bg-gray-900 ia:text-gray-900/60 ia:dark:text-white/60 ia:px-3 ia:py-1.5 ia:-ml-px ia:text-xs ia:font-medium ia:rounded-t-lg ia:border ia:border-gray-200 ia:dark:border-gray-800 ia:border-b-0 ia:cursor-pointer"
            tabindex="-1"
            type="button"
          >
            CSS Property Inspector 
            <svg
              class="lucide lucide-chevron-down ia:size-4 ia:text-gray-500 ia:dark:text-gray-400 ia:flex-shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
        </div>
        <div
          class="ia:relative ia:h-[400px]"
        >
          <div
            class="ia:relative ia:overflow-hidden ia:h-full ia:pr-3"
            dir="ltr"
            style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
          >
            <style>
              [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
            </style>
            <div
              class="ia:h-full ia:w-full ia:rounded-[inherit]"
              data-radix-scroll-area-viewport=""
              style="overflow-x: hidden; overflow-y: scroll;"
            >
              <div
                style="min-width: 100%; display: table;"
              >
                <div
                  class="ia:space-y-1"
                >
                  <div
                    class="ia:w-full"
                  >
                    <button
                      class="ia:sticky ia:top-0 ia:z-10 ia:bg-white ia:dark:bg-gray-900 ia:w-full ia:px-4 ia:py-2 ia:flex ia:items-center ia:gap-2 ia:cursor-pointer ia:hover:bg-gray-100 ia:dark:hover:bg-gray-800 ia:text-left"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-chevron-down ia:size-4 ia:text-gray-500 ia:dark:text-gray-400 ia:flex-shrink-0"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m6 9 6 6 6-6"
                        />
                      </svg>
                      <svg
                        class="lucide lucide-paintbrush ia:size-4 ia:text-emerald-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m14.622 17.897-10.68-2.913"
                        />
                        <path
                          d="M18.376 2.622a1 1 0 1 1 3.002 3.002L17.36 9.643a.5.5 0 0 0 0 .707l.944.944a2.41 2.41 0 0 1 0 3.408l-.944.944a.5.5 0 0 1-.707 0L8.354 7.348a.5.5 0 0 1 0-.707l.944-.944a2.41 2.41 0 0 1 3.408 0l.944.944a.5.5 0 0 0 .707 0z"
                        />
                        <path
                          d="M9 8c-1.804 2.71-3.97 3.46-6.583 3.948a.507.507 0 0 0-.302.819l7.32 8.883a1 1 0 0 0 1.185.204C12.735 20.405 16 16.792 16 15"
                        />
                      </svg>
                      <h3
                        class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                      >
                        Tokens
                      </h3>
                      <div
                        class="ia:inline-flex ia:items-center ia:rounded-full ia:border ia:px-2.5 ia:py-0.5 ia:text-xs ia:font-semibold ia:transition-colors ia:focus:outline-none ia:focus:ring-2 ia:focus:ring-ring ia:focus:ring-offset-2 ia:ml-auto ia:border-gray-200 ia:dark:border-gray-700 ia:text-gray-700 ia:dark:text-gray-300"
                      >
                        1
                      </div>
                    </button>
                    <div
                      class="ia:space-y-2 ia:px-4 ia:py-2"
                    >
                      <div
                        class="ia:rounded-md ia:border ia:border-gray-200 ia:dark:border-gray-800 ia:p-2 ia:text-sm ia:bg-white ia:dark:bg-gray-900"
                      >
                        <div
                          class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                        >
                          color
                        </div>
                        <div
                          class="ia:grid ia:grid-cols-2 ia:gap-x-2 ia:gap-y-1 ia:mt-1 ia:text-xs"
                        >
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Value:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <div
                              class="ia:size-3.5 ia:rounded-sm ia:border ia:border-gray-300 ia:dark:border-gray-700 ia:flex-shrink-0"
                              data-testid="color-swatch"
                              style="background-color: red; box-shadow: inset 0 0 0 1px rgba(0,0,0,0.05);"
                            />
                            <span
                              class="ia:font-mono ia:flex-1 ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              red
                            </span>
                          </div>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Token:
                          </span>
                          <span
                            class="ia:font-mono ia:text-emerald-600 ia:dark:text-emerald-500"
                          >
                            COLOR_PRIMARY
                          </span>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Computed:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              rgb(255, 0, 0)
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ia:shrink-0 ia:h-[1px] ia:w-full ia:my-1 ia:bg-gray-200 ia:dark:bg-gray-800"
                    data-orientation="horizontal"
                    role="none"
                  />
                  <div
                    class="ia:w-full"
                  >
                    <button
                      class="ia:sticky ia:top-0 ia:z-10 ia:bg-white ia:dark:bg-gray-900 ia:w-full ia:px-4 ia:py-2 ia:flex ia:items-center ia:gap-2 ia:cursor-pointer ia:hover:bg-gray-100 ia:dark:hover:bg-gray-800 ia:text-left"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-chevron-down ia:size-4 ia:text-gray-500 ia:dark:text-gray-400 ia:flex-shrink-0"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m6 9 6 6 6-6"
                        />
                      </svg>
                      <svg
                        class="lucide lucide-variable ia:size-4 ia:text-violet-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 21s-4-3-4-9 4-9 4-9"
                        />
                        <path
                          d="M16 3s4 3 4 9-4 9-4 9"
                        />
                        <line
                          x1="15"
                          x2="9"
                          y1="9"
                          y2="15"
                        />
                        <line
                          x1="9"
                          x2="15"
                          y1="9"
                          y2="15"
                        />
                      </svg>
                      <h3
                        class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                      >
                        Variables
                      </h3>
                      <div
                        class="ia:inline-flex ia:items-center ia:rounded-full ia:border ia:px-2.5 ia:py-0.5 ia:text-xs ia:font-semibold ia:transition-colors ia:focus:outline-none ia:focus:ring-2 ia:focus:ring-ring ia:focus:ring-offset-2 ia:ml-auto ia:border-gray-200 ia:dark:border-gray-700 ia:text-gray-700 ia:dark:text-gray-300"
                      >
                        1
                      </div>
                    </button>
                    <div
                      class="ia:space-y-2 ia:px-4 ia:py-2"
                    >
                      <div
                        class="ia:rounded-md ia:border ia:border-gray-200 ia:dark:border-gray-800 ia:p-2 ia:text-sm ia:bg-white ia:dark:bg-gray-900"
                      >
                        <div
                          class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                        >
                          background-color
                        </div>
                        <div
                          class="ia:grid ia:grid-cols-2 ia:gap-x-2 ia:gap-y-1 ia:mt-1 ia:text-xs"
                        >
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Value:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              var(--bg-color)
                            </span>
                          </div>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Variable:
                          </span>
                          <span
                            class="ia:font-mono ia:text-violet-600 ia:dark:text-violet-500"
                          >
                            --bg-color
                          </span>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Variable Value:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              rgb(0, 0, 0)
                            </span>
                          </div>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Computed:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              rgb(0, 0, 0)
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ia:shrink-0 ia:h-[1px] ia:w-full ia:my-1 ia:bg-gray-200 ia:dark:bg-gray-800"
                    data-orientation="horizontal"
                    role="none"
                  />
                  <div
                    class="ia:w-full"
                  >
                    <button
                      class="ia:sticky ia:top-0 ia:z-10 ia:bg-white ia:dark:bg-gray-900 ia:w-full ia:px-4 ia:py-2 ia:flex ia:items-center ia:gap-2 ia:cursor-pointer ia:hover:bg-gray-100 ia:dark:hover:bg-gray-800 ia:text-left"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-chevron-down ia:size-4 ia:text-gray-500 ia:dark:text-gray-400 ia:flex-shrink-0"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m6 9 6 6 6-6"
                        />
                      </svg>
                      <svg
                        class="lucide lucide-code ia:size-4 ia:text-sky-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="16 18 22 12 16 6"
                        />
                        <polyline
                          points="8 6 2 12 8 18"
                        />
                      </svg>
                      <h3
                        class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                      >
                        Computed
                      </h3>
                      <div
                        class="ia:inline-flex ia:items-center ia:rounded-full ia:border ia:px-2.5 ia:py-0.5 ia:text-xs ia:font-semibold ia:transition-colors ia:focus:outline-none ia:focus:ring-2 ia:focus:ring-ring ia:focus:ring-offset-2 ia:ml-auto ia:border-gray-200 ia:dark:border-gray-700 ia:text-gray-700 ia:dark:text-gray-300"
                      >
                        1
                      </div>
                    </button>
                    <div
                      class="ia:space-y-2 ia:px-4 ia:py-2"
                    >
                      <div
                        class="ia:rounded-md ia:border ia:border-gray-200 ia:dark:border-gray-800 ia:p-2 ia:text-sm ia:bg-white ia:dark:bg-gray-900"
                      >
                        <div
                          class="ia:font-medium ia:text-gray-900 ia:dark:text-white"
                        >
                          font-size
                        </div>
                        <div
                          class="ia:grid ia:grid-cols-2 ia:gap-x-2 ia:gap-y-1 ia:mt-1 ia:text-xs"
                        >
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Value:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              16px
                            </span>
                          </div>
                          <span
                            class="ia:text-gray-500 ia:dark:text-gray-400"
                          >
                            Computed:
                          </span>
                          <div
                            class="ia:flex ia:items-center ia:gap-1.5"
                          >
                            <span
                              class="ia:font-mono  ia:text-gray-900 ia:dark:text-gray-100"
                            >
                              16px
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
